### ISSUE TITLE
ReadOnly builders must have a validation method

### ISSUE THREAD

**Issue by octonato** [lagom/persistence-api-experiments#3]
As explained by @ignasi35  here: https://github.com/lagom/persistence-api-experiments/pull/2#pullrequestreview-65852677

**Comment by ignasi35** [comment]
Please, review my further comments on that review. I don't think `validation` should be a separate concern.

**Comment by octonato** [comment]
Yes, that was my point for Command Handlers that do emit events.

That is not valid for a `ReadOnly`. We may need something that block the execution of the command, hence the `validate` method. 

We can't do it in the `replyWith` method because we only have access to the `State`, not the `Command`. Otherwise we could throw an exception in the `replyWith`.
